In [1]:
import pandas as pd
import numpy as np



In [4]:
data=pd.read_csv('C:/Users/aditya/Desktop/Internship/Datasets/data1_nonull.csv')

In [5]:
data.head()

Unnamed: 0 Country Name Country Code Counterpart-Area Name  \
0         110  Afghanistan          AFG                 World   
1         115  Afghanistan          AFG                 World   
2         338  Afghanistan          AFG                 World   
3         377  Afghanistan          AFG                 World   
4         389  Afghanistan          AFG                 World   

  Counterpart-Area Code                                        Series Name  \
0                   WLD  Disbursements on external debt, long-term (DIS...   
1                   WLD  Disbursements on external debt, public and pub...   
2                   WLD                  PPG, bilateral (DIS, current US$)   
3                   WLD               PPG, multilateral (DIS, current US$)   
4                   WLD         PPG, official creditors (DIS, current US$)   

      Series Code  1970  1971  1972  ...        2019      2020        2021  \
0  DT.DIS.DLXF.CD   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2   
1  DT.DIS.DPPG.CD   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2   
2  DT.DIS.BLAT.CD   0.0   0.0   0.0  ...  27851000.5       0.0  81807974.7   
3  DT.DIS.MLAT.CD   0.0   0.0   0.0  ...    739200.6  440209.2   9717606.5   
4  DT.DIS.OFFT.CD   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2   

          2022        2023        2024        2025        2026       2027  \
0  109228850.6  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5   
1  109228850.6  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5   
2   86924699.4  59607890.3  37990151.3  21443526.5   9922365.3  6005463.7   
3   22304151.2  16529884.4  11755759.3   7837172.9   5360333.5  3063047.8   
4  109228850.6  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5   

        2028  
0  3491523.9  
1  3491523.9  
2  1960000.0  
3  1531523.9  
4  3491523.9  

[5 rows x 66 columns]

In [6]:
a = data['Series Code'].unique()

In [7]:
len(a)

127

In [8]:
data.shape

(5635, 66)

# 6. AVERAGE AMOUNT OF DEBT ACROSS INDICATORS

In [9]:
data.drop(['2022', '2023', '2024', '2025','2026', '2027', '2028'],inplace=True,axis=1)

In [10]:
data.columns

Index(['Unnamed: 0', 'Country Name', 'Country Code', 'Counterpart-Area Name',
       'Counterpart-Area Code', 'Series Name', 'Series Code', '1970', '1971',
       '1972', '1973', '1974', '1975', '1976', '1977', '1978', '1979', '1980',
       '1981', '1982', '1983', '1984', '1985', '1986', '1987', '1988', '1989',
       '1990', '1991', '1992', '1993', '1994', '1995', '1996', '1997', '1998',
       '1999', '2000', '2001', '2002', '2003', '2004', '2005', '2006', '2007',
       '2008', '2009', '2010', '2011', '2012', '2013', '2014', '2015', '2016',
       '2017', '2018', '2019', '2020', '2021'],
      dtype='object')

In [11]:
data_2 = data.select_dtypes('float64')

In [12]:
data_2.head()

1970  1971  1972  1973  1974  1975  1976  1977  1978  1979  ...  \
0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   
1   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   
2   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   
3   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   
4   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   

         2012        2013        2014        2015        2016       2017  \
0  52059123.0  36656261.1  31740710.4  26037493.8  11669646.0  3241563.2   
1  52059123.0  36656261.1  31740710.4  26037493.8  11669646.0  3241563.2   
2         0.0         0.0  10120533.3  17992800.0   1438712.0  2082133.3   
3  52059123.0  36656261.1  21620177.1   8044693.8  10230934.0  1159429.9   
4  52059123.0  36656261.1  31740710.4  26037493.8  11669646.0  3241563.2   

         2018        2019      2020        2021  
0  16323196.4  28590201.1  440209.2  91525581.2  
1  16323196.4  28590201.1  440209.2  91525581.2  
2  10346133.3  27851000.5       0.0  81807974.7  
3   5977063.1    739200.6  440209.2   9717606.5  
4  16323196.4  28590201.1  440209.2  91525581.2  

[5 rows x 52 columns]

In [13]:
data['Sum'] = data_2.sum(axis = 1)

In [14]:
data.head()

Unnamed: 0 Country Name Country Code Counterpart-Area Name  \
0         110  Afghanistan          AFG                 World   
1         115  Afghanistan          AFG                 World   
2         338  Afghanistan          AFG                 World   
3         377  Afghanistan          AFG                 World   
4         389  Afghanistan          AFG                 World   

  Counterpart-Area Code                                        Series Name  \
0                   WLD  Disbursements on external debt, long-term (DIS...   
1                   WLD  Disbursements on external debt, public and pub...   
2                   WLD                  PPG, bilateral (DIS, current US$)   
3                   WLD               PPG, multilateral (DIS, current US$)   
4                   WLD         PPG, official creditors (DIS, current US$)   

      Series Code  1970  1971  1972  ...        2013        2014        2015  \
0  DT.DIS.DLXF.CD   0.0   0.0   0.0  ...  36656261.1  31740710.4  26037493.8   
1  DT.DIS.DPPG.CD   0.0   0.0   0.0  ...  36656261.1  31740710.4  26037493.8   
2  DT.DIS.BLAT.CD   0.0   0.0   0.0  ...         0.0  10120533.3  17992800.0   
3  DT.DIS.MLAT.CD   0.0   0.0   0.0  ...  36656261.1  21620177.1   8044693.8   
4  DT.DIS.OFFT.CD   0.0   0.0   0.0  ...  36656261.1  31740710.4  26037493.8   

         2016       2017        2018        2019      2020        2021  \
0  11669646.0  3241563.2  16323196.4  28590201.1  440209.2  91525581.2   
1  11669646.0  3241563.2  16323196.4  28590201.1  440209.2  91525581.2   
2   1438712.0  2082133.3  10346133.3  27851000.5       0.0  81807974.7   
3  10230934.0  1159429.9   5977063.1    739200.6  440209.2   9717606.5   
4  11669646.0  3241563.2  16323196.4  28590201.1  440209.2  91525581.2   

           Sum  
0  935116905.3  
1  935116905.3  
2  198840347.3  
3  736276558.0  
4  935116905.3  

[5 rows x 60 columns]

In [15]:
d = pd.DataFrame(data.groupby('Series Code')['Sum'].sum())
d.head(30)

Sum
Series Code                      
DT.AMT.BLAT.CD       4.530176e+12
DT.AMT.BLAT.GG.CD    3.267616e+12
DT.AMT.BLAT.OPS.CD   1.084300e+12
DT.AMT.BLAT.PRVG.CD  1.216291e+10
DT.AMT.BLAT.PS.CD    4.509988e+12
DT.AMT.BLTC.CD       4.001069e+11
DT.AMT.DEGG.CD       1.380930e+13
DT.AMT.DLXF.CD       5.661296e+13
DT.AMT.DOPS.CD       8.782869e+12
DT.AMT.DPNG.CD       3.227120e+13
DT.AMT.DPPG.CD       2.326073e+13
DT.AMT.MLAT.CD       4.413790e+12
DT.AMT.MLAT.GG.CD    3.443575e+12
DT.AMT.MLAT.OPS.CD   7.928563e+11
DT.AMT.MLAT.PRVG.CD  3.254720e+10
DT.AMT.MLAT.PS.CD    4.371263e+12
DT.AMT.MLTC.OPS.CD   1.465494e+10
DT.AMT.MLTC.PS.CD    4.182991e+11
DT.AMT.OFFT.CD       9.077156e+12
DT.AMT.OFFT.GG.CD    6.874004e+12
DT.AMT.OFFT.OPS.CD   1.906804e+12
DT.AMT.OFFT.PRVG.CD  5.131349e+10
DT.AMT.OFFT.PS.CD    9.013477e+12
DT.AMT.PBND.CD       4.943487e+12
DT.AMT.PBND.GG.CD    3.646211e+12
DT.AMT.PBND.OPS.CD   1.021469e+12
DT.AMT.PBND.PS.CD    4.942445e+12
DT.AMT.PCBK.CD       5.606434e+12
DT.AMT.PCBK.GG.CD    1.456473e+12
DT.AMT.PCBK.OPS.CD   3.441029e+12

### ABOVE DATAFRAME SHOWS AVERAGE DEBT ACROSS INDICATORS

# 7. The highest amount of principal repayments

In [16]:
pr = d.sort_values(['Sum'], ascending=[False])
pr.head()

Sum
Series Code                 
DT.DIS.DLXF.CD  8.760722e+13
DT.TDS.DECT.CD  8.278910e+13
DT.TDS.DLXF.CD  7.566665e+13
DT.AMT.DLXF.CD  5.661296e+13
DT.DIS.DPPG.CD  4.122165e+13

In [17]:
pr.head(10)

Sum
Series Code                 
DT.DIS.DLXF.CD  8.760722e+13
DT.TDS.DECT.CD  8.278910e+13
DT.TDS.DLXF.CD  7.566665e+13
DT.AMT.DLXF.CD  5.661296e+13
DT.DIS.DPPG.CD  4.122165e+13
DT.TDS.DPNG.CD  4.002073e+13
DT.TDS.DPPG.CD  3.504860e+13
DT.TDS.DEPS.CD  3.483759e+13
DT.TDS.PNGC.CD  3.366918e+13
DT.AMT.DPNG.CD  3.227120e+13

In [18]:
pr = pr.reset_index()
pr.head()

Series Code           Sum
0  DT.DIS.DLXF.CD  8.760722e+13
1  DT.TDS.DECT.CD  8.278910e+13
2  DT.TDS.DLXF.CD  7.566665e+13
3  DT.AMT.DLXF.CD  5.661296e+13
4  DT.DIS.DPPG.CD  4.122165e+13

In [19]:
s = pr[ (pr["Series Code"] == 'DT.AMT.DLTF.CD') | (pr["Series Code"] == 'DT.AMT.DEPS.CD')]
s

Empty DataFrame
Columns: [Series Code, Sum]
Index: []

In [20]:
r = pr[ (pr["Series Code"] == 'DT.AMT.DEGG.CD') | (pr["Series Code"] == 'DT.AMT.DLXF.CD')
       | (pr["Series Code"] == 'DT.AMT.DLTF.CD') | (pr["Series Code"] == 'DT.AMT.DOPS.CD') 
       | (pr["Series Code"] == 'DT.AMT.PRPG.CD') | (pr["Series Code"] == 'DT.AMT.DPNG.CD')
       | (pr["Series Code"] == 'DT.AMT.DPPG.CD') | (pr["Series Code"] == 'DT.AMT.DEPS.CD')]
r.head(8)


Series Code           Sum
3    DT.AMT.DLXF.CD  5.661296e+13
9    DT.AMT.DPNG.CD  3.227120e+13
11   DT.AMT.DPPG.CD  2.326073e+13
18   DT.AMT.DEGG.CD  1.380930e+13
31   DT.AMT.DOPS.CD  8.782869e+12
113  DT.AMT.PRPG.CD  1.351307e+11

In [21]:
r.sort_values(['Sum'], ascending=[False])
r

Series Code           Sum
3    DT.AMT.DLXF.CD  5.661296e+13
9    DT.AMT.DPNG.CD  3.227120e+13
11   DT.AMT.DPPG.CD  2.326073e+13
18   DT.AMT.DEGG.CD  1.380930e+13
31   DT.AMT.DOPS.CD  8.782869e+12
113  DT.AMT.PRPG.CD  1.351307e+11

### DT.AMT.DLXF.CD IS THE DEBT INDICATOR WITH THE HIGHEST AMOUNT OF PRINCIPAL REPAYMENT OF 5.661296E+13 DOLLARS

# 8. The most common debt indicator

In [22]:
a = data['Series Code'].value_counts()
a.head(30)

DT.DIS.DPPG.CD       128
DT.DIS.DLXF.CD       128
DT.DIS.OFFT.CD       128
DT.DIS.MLAT.CD       126
DT.TDS.DECT.CD        93
DT.TDS.DPPG.CD        89
DT.TDS.DEGG.CD        89
DT.TDS.DLXF.CD        89
DT.TDS.DEPS.CD        89
DT.INT.DEPS.CD        88
DT.INT.DLXF.CD        88
DT.INT.DPPG.CD        88
DT.INT.DEGG.CD        88
DT.INT.OFFT.GG.CD     87
DT.INT.OFFT.CD        87
DT.TDS.OFFT.GG.CD     87
DT.TDS.OFFT.PS.CD     87
DT.INT.OFFT.PS.CD     87
DT.AMT.DPPG.CD        85
DT.AMT.DLXF.CD        85
DT.AMT.OFFT.CD        82
DT.AMT.DEGG.CD        82
DT.AMT.OFFT.PS.CD     82
DT.TDS.BLAT.CD        80
DT.TDS.BLAT.PS.CD     80
DT.AMT.OFFT.GG.CD     79
DT.INT.BLAT.PS.CD     78
DT.INT.BLAT.CD        78
DT.TDS.BLAT.GG.CD     75
DT.TDS.MLAT.PS.CD     75
Name: Series Code, dtype: int64

### DT.DIS.DLXF.CD, DT.DIS.DPPG.CD and DT.DIS.OFFT.CD are the 3 most common debt indicators

# 9. Other viable debt issues and conclusion


### VIABLE DEBT ISSUES AND CONCLUSIONS :
#### * THERE ARE 132 UNIQUE COUNTRIES IN TOTAL AND 127 DISTINCT DEBT INDICATORS
#### * WHERE LOW AND MIDDLE INCOME COUNTRIES HAVE THE HIGEST DEBT (WITH GROUPING)
#### * AND BRAZIL AS A COUNTRY HAS THE HIGEST DEBT (WITHOUT GROUPING)
#### * DT.AMT.DLXF.CD IS THE DEBT INDICATOR WITH THE HIGHEST AMOUNT OF PRINCIPAL REPAYMENT OF 5.661296E+13 DOLLARS
#### * DT.DIS.DLXF.CD, DT.DIS.DPPG.CD AND DT.DIS.OFFT.CD ARE THE 3 MOST COMMON DEBT INDICATORS


# PYLOTS  FOR DEBT INDICATORS

In [23]:
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
from plotly.offline import iplot
import plotly as py
import cufflinks as cf
py.offline.init_notebook_mode(connected=True)
cf.go_offline()

## PLOT FOR TOP 10 DEBT INDICATORS

In [24]:
c = pr[0:10]

In [25]:
c.iplot(kind='bar', x = 'Series Code',y='Sum',title='TOP 10 DEBT INDICATORS WITH TOTAL DEBT AMOUNTS', colors = 'yellow')

## PLOT FOR MOST COMMON DEBT INDICATORS WITH THEIR TOTAL AMOUNT OF DEBT

In [26]:
p = [['DT.DIS.DLXF.CD', 8.760722e+13 ], ['DT.DIS.DPPG.CD', 4.122165e+13 ], ['DT.DIS.OFFT.CD', 1.563509e+13 ],
     ['DT.DIS.MLAT.CD', 9.091839e+12 ]]
df = pd.DataFrame(p, columns = ['Debt Indicator', 'Debt Amount'])
df

Debt Indicator   Debt Amount
0  DT.DIS.DLXF.CD  8.760722e+13
1  DT.DIS.DPPG.CD  4.122165e+13
2  DT.DIS.OFFT.CD  1.563509e+13
3  DT.DIS.MLAT.CD  9.091839e+12

In [27]:
df.iplot(kind='bar',x = 'Debt Indicator', y='Debt Amount',title='MOST COMMON DEBT INDICATORS WITH THEIR TOTAL DEBT AMOUNTS',
        colors = 'cyan')

## PLOT FOR PRINCIPAL REPAYMENT DEBT INDICATORS

In [28]:
r = pr[ (pr["Series Code"] == 'DT.AMT.DEGG.CD') | (pr["Series Code"] == 'DT.AMT.DLXF.CD')
       | (pr["Series Code"] == 'DT.AMT.DLTF.CD') | (pr["Series Code"] == 'DT.AMT.DOPS.CD') 
       | (pr["Series Code"] == 'DT.AMT.PRPG.CD') | (pr["Series Code"] == 'DT.AMT.DPNG.CD')
       | (pr["Series Code"] == 'DT.AMT.DPPG.CD') | (pr["Series Code"] == 'DT.AMT.DEPS.CD')]
r.head(8)


Series Code           Sum
3    DT.AMT.DLXF.CD  5.661296e+13
9    DT.AMT.DPNG.CD  3.227120e+13
11   DT.AMT.DPPG.CD  2.326073e+13
18   DT.AMT.DEGG.CD  1.380930e+13
31   DT.AMT.DOPS.CD  8.782869e+12
113  DT.AMT.PRPG.CD  1.351307e+11

In [29]:
r.iplot(kind='bar', x = 'Series Code',y='Sum',title='PRINCIPAL REPAYMENT DEBT INDICATORS WITH TOTAL DEBT AMOUNTS', 
        colors = 'RGB(255,0,0)')